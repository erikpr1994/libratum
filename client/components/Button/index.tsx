import { useRouter } from 'next/router';

import { shadow, colors } from '../../styles/theme.js';

type ButtonType = {
  name: string;
  width: number;
  height: number;
  color: string;
  onClick?: Function;
};

export default function Button({
  name,
  width,
  height,
  color,
  onClick,
}: ButtonType) {
  const router = useRouter();
  return (
    <>
      <button onClick={() => router.replace(`/dashboard`)} id="test">
        <p>{name}</p>
      </button>
      <style jsx>{`
        button {
          width: ${width}px;
          height: ${height}px;
          display: flex;
          justify-content: center;
          align-items: center;
          box-shadow: ${shadow};
          border-radius: 15px;
          background: ${color};
          color: ${colors.white};
          cursor: pointer;
        }
      `}</style>
    </>
  );
}
