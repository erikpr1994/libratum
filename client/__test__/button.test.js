import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Button from '../components/Button';

afterEach(cleanup);

describe('Button', () => {
  const defaultProps = {
    name: 'Rebalance',
  };
  test('Button renders with correct text', () => {
    const { queryByText, container } = render(<Button {...defaultProps} />);
    expect(queryByText('Rebalance')).toBeTruthy();
    expect(container.baseElement).toMatchSnapshot();
  });
});
