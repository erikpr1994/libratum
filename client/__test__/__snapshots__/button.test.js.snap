// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders with correct text 1`] = `undefined`;
